### ISSUE TITLE
Group by query get's incorrectly parsed

### ISSUE THREAD

**Issue by wschella** [joachimvh/SPARQLAlgebra.js#37]
The query below

```sparql
SELECT ?contains (sample(?object) as ?gc)
WHERE {
  ?subject ?predicate ?object
} GROUP BY (contains(str(?subject), "Publikationen"))
```

misses all the information in relation to the expression (contains).

Output:

```json
{
    "type": "project",
    "input": {
        "type": "extend",
        "input": {
            "type": "group",
            "input": {
                "type": "bgp",
                "patterns": [
                    {
                        "subject": {
                            "value": "subject"
                        },
                        "predicate": {
                            "value": "predicate"
                        },
                        "object": {
                            "value": "object"
                        },
                        "graph": {
                            "value": ""
                        },
                        "type": "pattern"
                    }
                ]
            },
            "variables": [
                {
                    "value": ""
                }
            ],
            "aggregates": [
                {
                    "type": "expression",
                    "expressionType": "aggregate",
                    "aggregator": "sample",
                    "expression": {
                        "type": "expression",
                        "expressionType": "term",
                        "term": {
                            "value": "object"
                        }
                    },
                    "distinct": false,
                    "variable": {
                        "value": "var0"
                    }
                }
            ]
        },
        "variable": {
            "value": "gc"
        },
        "expression": {
            "type": "expression",
            "expressionType": "term",
            "term": {
                "value": "var0"
            }
        }
    },
    "variables": [
        {
            "value": "contains"
        },
        {
            "value": "gc"
        }
    ]
}
```

**Comment by wschella** [comment]
A simpler example might be `SELECT * WHERE {?x ?y ?z} GROUP BY str(?x)`, where the notion of `str` just dissapears.

**Comment by joachimvh** [comment]
Apparently there was a problem with GROUP BY in this library. Should be fixed in 3d7a83acecddb8689961bd6ef630ed25ce1bbc3e
